src/orange/OrangeTrackView.hh

```diff
@@ -474,9 +514,14 @@ CELER_FUNCTION void OrangeTrackView::move_internal(real_type dist)
  */
 CELER_FUNCTION void OrangeTrackView::move_internal(Real3 const& pos)
 {
-    auto lsa = this->make_lsa();
-    lsa.pos() = pos;
-    lsa.surf() = LocalSurfaceId{};
+    for (auto i : range(states_.level[track_slot_] + 1))
+    {
+        auto lsa = this->make_lsa(LevelId{i});
+        lsa.pos() = pos;
```

Comment:
`pos` here is global, and we need to propagate it down to each level. I suspect it would be "safer" (more robust against roundoff error if the deepest level is orders of magnitude displaced from the global level) to do that by updating each level by accumulating `pos - this->pos()`, but the simpler thing to do would be doing repeated global-to-local transformations like we do with the initialize function.